File: LEBOv2/lebo/src/shared/types/gameData.ts
export interface GameNode {
  id: string
  name: string
  pointCost: number
  maxPoints: number
  prerequisiteNodeIds: string[]
  effectDescription: string
  tags: string[]
  position: { x: number; y: number }
  size: 'small' | 'medium' | 'large'
}

export interface MasteryData {
  masteryId: string
  masteryName: string
  nodes: Record<string, GameNode>
}

export interface ClassData {
  classId: string
  className: string
  baseTree: Record<string, GameNode>
  masteries: Record<string, MasteryData>
}

export interface GameDataManifest {
  schemaVersion: number
  gameVersion: string
  dataVersion: string
  generatedAt: string
  classes: string[]
}

export interface GameData {
  manifest: GameDataManifest
  classes: Record<string, ClassData>
}

export interface DataVersionCheckResult {
  isStale: boolean
  localVersion: string
  remoteVersion: string
  versionsBehind: number
}

File: LEBOv2/lebo/src-tauri/src/models/game_data.rs
use serde::{Deserialize, Serialize};

#[derive(Debug, Deserialize, Serialize, Clone)]
pub struct NodeEffect {
    pub description: String,
    pub tags: Vec<String>,
}

#[derive(Debug, Deserialize, Serialize, Clone)]
#[serde(rename_all = "camelCase")]
pub struct RawGameNode {
    pub id: String,
    pub name: String,
    pub x: f64,
    pub y: f64,
    pub size: String,
    pub max_points: u32,
    pub effects: Vec<NodeEffect>,
}

#[derive(Debug, Deserialize, Serialize, Clone)]
#[serde(rename_all = "camelCase")]
pub struct RawEdge {
    pub from_id: String,
    pub to_id: String,
}

#[derive(Debug, Deserialize, Serialize, Clone)]
pub struct RawTreeData {
    pub nodes: Vec<RawGameNode>,
    pub edges: Vec<RawEdge>,
}

#[derive(Debug, Deserialize, Serialize, Clone)]
#[serde(rename_all = "camelCase")]
pub struct RawMastery {
    pub id: String,
    pub name: String,
    pub passive_tree: RawTreeData,
}

#[derive(Debug, Deserialize, Serialize, Clone)]
#[serde(rename_all = "camelCase")]
pub struct RawClassData {
    pub id: String,
    pub name: String,
    pub base_tree: RawTreeData,
    pub masteries: Vec<RawMastery>,
}

#[derive(Debug, Deserialize, Serialize, Clone)]
#[serde(rename_all = "camelCase")]
pub struct GameDataManifest {
    pub schema_version: u32,
    pub game_version: String,
    pub data_version: String,
    pub generated_at: String,
    pub classes: Vec<String>,
}

#[derive(Debug, Serialize, Deserialize, Clone)]
#[serde(rename_all = "camelCase")]
pub struct DataVersionCheckResult {
    pub is_stale: bool,
    pub local_version: String,
    pub remote_version: String,
    pub versions_behind: u32,
}
